plugins/outputs/application_insights/application_insights.go

```
@@ -0,0 +1,422 @@
+}
+
+type ApplicationInsights struct {
+	InstrumentationKey      string
+	Timeout                 internal.Duration
+	EnableDiagnosticLogging bool
+	ContextTagSources       map[string]string
+	diagMsgSubscriber       DiagnosticsMessageSubscriber
+	transmitter             TelemetryTransmitter
+	diagMsgListener         appinsights.DiagnosticsMessageListener
+}
+
+var (
+	sampleConfig = `
+## Instrumentation key of the Application Insights resource.
+instrumentationKey = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx"
+
+## Timeout on close. If not provided, will default to 5s. 0s means no timeout (not recommended).
+# timeout = "5s"
+
+## Determines whether diagnostic logging (for Application Insights endpoint traffic) is enabled. Default is false.
+# enableDiagnosticLogging = "true"
+
+## ContextTagSources dictionary instructs the Application Insights plugin to set Application Insights context tags using metric properties.
+## In this dictionary keys are Application Insights context tags to set, and values are names of metric properties to use as source of data.
+## For example:
+# [outputs.application_insights.contextTagSources]
+# "ai.cloud.role" = "kubernetes_container_name"
+# "ai.cloud.roleInstance" = "kubernetes_pod_name"
+## will set the ai.cloud.role context tag to the value of kubernetes_container_name property (if present), 
+## and the ai.cloud.roleInstance context tag to the value of kubernetes_pod_name property.
+## For list of all context tag keys see https://github.com/Microsoft/ApplicationInsights-Go/blob/master/appinsights/contracts/contexttagkeys.go
+`
+	is32Bit        bool
+	is32BitChecked bool
+)
+
+func (a *ApplicationInsights) SampleConfig() string {
+	return sampleConfig
+}
+
+func (a *ApplicationInsights) Description() string {
+	return "Send telegraf metrics to Azure Application Insights"
+}
+
+func (a *ApplicationInsights) Connect() error {
+	if a.transmitter == nil && a.InstrumentationKey != "" {
```

If the instrumentation key is unset, should we return an error?